exodus_lambda/functions/origin_request.py

```diff
@@ -103,10 +102,9 @@ def handler(self, event, context):
                 )
                 content_type = query_result["Items"][0]["content_type"]["S"]
                 if content_type:
-                    query = urllib.parse.urlencode(
-                        {"ResponseContentType": content_type}
+                    request["querystring"] = (
+                        "response-content-type=%s" % content_type
```

Comment:
Sorry to keep on the same issue here, but the lack of encoding here still doesn't sit right with me.

Per the earlier discussion, I concluded that encoding is needed in querystring *in general* if we're dealing with arbitrary strings.

We aren't dealing with arbitrary strings, so it comes down to whether or not any content type might contain a character which needs encoding.  I went searching for relevant standards and couldn't convince myself that it's safe not to encode.  I found https://www.w3.org/Protocols/rfc1341/4_Content-Type.html

The standard says (roughly) that content types can contain any ASCII except one of:

```
()<>@,;:\"[]?.=
```

Notably: `%` is not included in the list of banned characters there, so a content type containing a literal `%` is apparently possible. If that happens then it doesn't seem likely to be interpreted by S3 correctly if it was not encoded.  I tried testing it but it's a little cumbersome.

Could you please have another try of making this encoded, so we are not laying a trap for our future selves?

You mentioned earlier that it causes tests to fail, if you still find this to be the case could you please share the code when it failed, and the output from the tests?